### ISSUE TITLE
very slow between models

### ISSUE THREAD

**Issue by andrewkern** [keras-team/autokeras#674]
Hi there-

autokeras seems to be extremely slow between models and will hang for minutes after reporting "No loss decrease after 5 epochs". The actually training epochs are moving much faster than the wait times between models. Is there any way I can speed this up?

thanks

**Comment by karolmajek** [comment]
same here with 4 GPUs and 20 cores.
In between, it is using a single core only to do ...?

**Comment by karolmajek** [comment]
Building from source when you have torch with cuda and TF-gpu without regular tensorflow solves this issue in my case.
`git clone` and `python setup.py install`

**Comment by wptoux** [comment]
It seems that the BayesianOptimizer relies on a multiprocess queue to check the training status,  whose method qsize, however,  is not reliable according to the python [document.](https://docs.python.org/3/library/multiprocessing.html). So the searching loop may not be able to break properly after training. I think this might be the issuse.

**Comment by AntonioDomenech** [comment]
I have the same issue, but in my case it is very curious because it actually goes faster in a 500€ laptop than a 2000€ tower computer with i9 processor and rtx 3080 graphics card.

Any one knows why?

laptop specs:
CPU - Core i5-8300H
Graphics card - NVIDIA GeForce GTX 1050 2GB
RAM - 4GB
memory - 1T ddr

Computer tower specs:
CPU - Core i9-10900kf
Graphics card - NVIDIA GeForce RTX 3080 10GB
memory - 1T ssd
RAM - 16GB

both have cuda installed and the training is faster in the computer, but the waiting time is much faster in the laptop, which in my onpinion is super wierd because the computer is better and faster than the laptop in any possible aspect.

**Comment by haifeng-jin** [comment]
This is really weird.

**Comment by AntonioDomenech** [comment]
I found this in the gpu support for tensorflow:

On systems with NVIDIA® Ampere GPUs (CUDA architecture 8.0) or newer, kernels are JIT-compiled from PTX and TensorFlow can take over 30 minutes to start up. This overhead can be limited to the first start up by increasing the default JIT cache size with: 'export CUDA_CACHE_MAXSIZE=2147483648' (see JIT Caching for details).

maybe that's the reason why it is slower in the case of the NVIDIA RTX 3080.

Anyway I'm not very experienced and I don't have a clue of what this JIT thing is or how can I change it...
